sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -1048,7 +1048,6 @@ private void validateConfig() {
             Preconditions.checkArgument(preferredRegions != null, "preferredRegions cannot be null when proactiveContainerInitConfig has been set");
             Preconditions.checkArgument(this.proactiveContainerInitConfig.getProactiveConnectionRegionsCount() <= this.preferredRegions.size(), "no. of regions to proactively connect to " +
                     "cannot be greater than the no.of preferred regions");
-            Preconditions.checkArgument(this.proactiveContainerInitConfig.getProactiveConnectionRegionsCount() > 0, "no. of proactive connection regions should be greater than 0");
```

Comment:
SHould we avoid negative values? I guess negative means on-demand connections only? Fine by me

Reply:
I think we should avoid negative values. Negative region configuration resulting in implicit disabling of proactive connection management is not very transparent.

Reply:
Agreed - so, why did you remove the check?

Reply:
Will merge for now - we can readd the check in subsequent PR

Reply:
When calling `CosmosProactiveContainerInitConfigBuilder#setProactiveConnectionRegionsCount` we have a negativity check. If this method is not called, the default region count is set to 1. A check in `CosmosClientBuilder` is redundant.